platform/server/elasticsearch/__tests__/ElasticsearchService.test.ts

```diff
@@ -0,0 +1,103 @@
+const mockCreateClient = jest.fn();
+
+jest.mock('elasticsearch', () => ({
+  Client: mockCreateClient
+}));
+
+import { ElasticsearchService } from '../ElasticsearchService';
+import { AdminClient } from '../AdminClient';
+import { ScopedDataClient } from '../ScopedDataClient';
+import { logger } from '../../../logging/__mocks__';
+import { k$, BehaviorSubject, first, toPromise } from '@elastic/kbn-observable';
+
+test('should not create multiple clients while service is running', async () => {
+  const createElasticsearchConfig = (type: string) => ({
+    filterHeaders: () => {},
+    toElasticsearchClientConfig: (options: any) => ({ type, options }),
+    requestHeadersWhitelist: []
+  });
+  const elasticsearchConfigs = {
+    forType: (type: string) => createElasticsearchConfig(type)
+  };
+  const configs$: any = new BehaviorSubject(elasticsearchConfigs);
+  const service = new ElasticsearchService(configs$, logger);
+
+  mockCreateClient.mockReset();
+
+  // Start subscribing to this.clients$,
+  // which means new elasticsearch data and admin clients are created,
+  // calling mockCreateClient once for each client (twice)
+  service.start();
```

Comment:
__optional nit:__ not a big deal right now, but both `service.start` and `service.stop` are marked as `async` and basically return `Promise`'s, so we may want to do `await` in the tests just to be on the safe side (in case we decide to do some "real async" stuff inside :)).

Reply:
thank you!